### ISSUE TITLE
Checking whether a wildcard contains an IP address not working

### ISSUE THREAD

**Issue by cmhdave** [seancfoley/IPAddress#36]
https://github.com/seancfoley/IPAddress/blob/2aa2e11b3d726503e058294ac8ba42d2d673b5bc/IPAddress/src/inet.ipaddr/inet/ipaddr/IPAddressString.java#L49

According to the above line, one should be able to test whether an IP address falls withing a wildcard range.  I've found that asterisks work but ranges do not work.  As an example here is code that can be tested in groovyConsole:
```
@Grab(group='com.github.seancfoley', module='ipaddress', version='5.2.1')

import inet.ipaddr.IPAddressString

IPAddressString myIp = new IPAddressString('108.31.1.1') 
IPAddressString ipRange = new IPAddressString('108.30-31.*.*') 

println ipRange.contains(myIp)
```

If you change `myIp` to `108.30.1.1` it returns `true`.  Expanding the range to `30-39` doesn't have any affect.

**Comment by seancfoley** [comment]
Yes, you are correct that the code above should return true.

I tried that out in Java with the latest code and it is ok there:
```java
IPAddressString myIp = new IPAddressString("108.31.1.1");
IPAddressString ipRange = new IPAddressString("108.30-31.*.*");
System.out.println(ipRange.contains(myIp));
System.out.println(ipRange.getAddress().contains(myIp.getAddress()));
```
```
true
true
```
So then I ran your groovy snippet in my Intellij Idea groovy console:
![image](https://user-images.githubusercontent.com/19226957/78401353-93b04a80-75c6-11ea-8a24-c7b6b45d43be.png)

It worked there too.

So then I checked out version 5.2.1 and got the same result using the intellij groovy console.

What "groovy console" are you referring to, is it something on the web?

**Comment by cmhdave** [comment]
Well that's fascinating.  Seeing your groovyConsole output I started thinking that on my Mac laptop I do a lot with Java 8 so I decided to try it again on my Windows laptop using Java 8 (latest Amazon Coretto 1.8.0.242) with Groovy 3.0.2 and it worked fine.  I'll check my Mac laptop's groovyConsole to see what combination of Groovy and Java it's failing on.  Hopefully that'll shed some light.

**Comment by seancfoley** [comment]
My wild guess is that the groovy console is interpreting the '*' to be its own wildcard and thus the code becomes something different.

**Comment by cmhdave** [comment]
I've done a little more debugging on my Mac laptop and I think I've figured out my problem.  I had a sample file with the code above saved to my desktop. It had version 4.3.3 of the IPAddress library.  I would run groovyConsole, load the file and run it... "false".  I then change the version to 5.2.1 in groovyConsole, run it... "false".

Next I shut down groovyConsole, restarted, set the version to 5.2.1 BEFORE I ran it... "true".  Change the version in groovyConsole to 4.3.3 and run it... "true".

So I believe that groovyConsole does NOT respect version changes in a @Grab command while the console is still running.  It has nothing to do with your library at all, sorry for the hassle!
